src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -5864,6 +5866,9 @@ private BoundLiteral BindLiteralConstant(LiteralExpressionSyntax node, BindingDi
                 type = GetSpecialType(specialType, diagnostics, node);
             }
 
+            if (node.Kind() is SyntaxKind.SingleLineRawStringLiteralExpression or SyntaxKind.MultiLineRawStringLiteralExpression)
```

Comment:
fwiw, we should never do that.  

1. that violates the rule that we should only issue syntactic errors for when we either are missing somethign required, or we have something extra we can't fit in the syntax model somehow.
2. it breaks incremental parsing.  incremental parsing cannot use nodes with diagnostics on it.  so thsi just ends up costing us.  

we should migrate any syntactic langver checks to be semantic.

Reply:
> fwiw, we should never do that.

I think this is probably news to the rest to the compiler team doing langversion checks, etc. We should probably have a tracking issue to remove unwanted parsing diagnostics under the "design debt" label.

Reply:
Here are the notes on guidelines we'd settled from previous discussion (including Neal and Cyrus). I don't see any direct mention of version checks. Rather we do want to avoid depending on parsing context to improve incremental parsing.
https://github.com/dotnet/roslyn/blob/main/docs/compilers/Design/Parser.md

Reply:
The rule is here:

> The syntax model defines the contract between syntax and semantic analysis. If the parser can build a meaningful syntax tree for a sequence of input tokens, then it should be left to semantic analysis to report when that tree is not semantically valid.

In these cases the lexer/parser can form meaningful trees.  So it's left to semantic analysis to let you know when something is not semantically ok.  A version check is a check on the semantics :)

Reply:
I don't think this rule means what you think it means. Producing a diagnostic during parsing is 100% compatible with having parsed a meaningful tree when we can.

Reply:
I wrote that rule :)  A diagnostic means that the tree is not valid in some fashion.  Indeed, that's precisely what the incremental parser uses to make this determination.  So the presence of a diagnostic saying "you're using the wrong lang version" ends up impacting all higher up versions which then interpret this as "there is a problem fitting the text to the syntax model".  

In this case this is purely a semantic check.  The tree is 100% syntactically valid (definitionally so), but there is an ambient external check that impacts it (specifically the idea of a lang-version).  This also violates the other syntactic rule i specified which is:

> The parser should, as much as possible, be context-free. 

this makes parsing not context free.  In order to parse you have to pass additional context in as it affects the nodes that you might produce.  This also means that if you change compiler, you have to reparse everything (which really doesn't make much sense since the text is all the same, and as such the syntactic structure is truly the same).

Where we have bent this a bit is with record-parsing, where there truly is a different in syntactic structure produced based on options.  But outside of that, where we generate teh same trees otherwise, putting version diagnostics in the parser violates these design principles.

Reply:
It is a valid scenario. Changing options is something users do, and it forces us to reparse everything because of these issues. This is another reason this is not a good idea.